unity-game/Assets/Scripts/ElevatorScripts/AutoElevator.cs

```diff
@@ -0,0 +1,220 @@
+            {
+                theEndRightBarrier.SetActive(false);
+            }
+        }
+
+
+
+        if (onEndPoint)
+        {
+            if (theStartPoint.gameObject.GetComponent<ElevatorBookends>().startLeftBarrier == true)
+            {
+                theStartLeftBarrier.SetActive(true);
+            }
+            else if (theStartPoint.gameObject.GetComponent<ElevatorBookends>().startLeftBarrier == false)
+            {
+                theStartLeftBarrier.SetActive(false);
+            }
+            if (theStartPoint.gameObject.GetComponent<ElevatorBookends>().startRightBarrier == true)
+            {
+                theStartRightBarrier.SetActive(true);
+            }
+            else if (theStartPoint.gameObject.GetComponent<ElevatorBookends>().startRightBarrier == false)
+            {
+                theStartRightBarrier.SetActive(false);
+            }
+        }
+
+
+
+        if (reachedDestination)
+        {
+            waitCounter = waitTime;
+            reachedDestination = false;
+        }
+
+
+
+        if (notMoving)
+        {
+            waitCounter -= Time.deltaTime;
+        }
+
+
+
+        if (!notMoving)
+        {
+            theLeftWall.SetActive(true);
+            theRightWall.SetActive(true);
+            if (theStartPoint.gameObject.GetComponent<ElevatorBookends>().startLeftBarrier == true)
+            {
+                theStartLeftBarrier.SetActive(true);
+            }
+            else if (theStartPoint.gameObject.GetComponent<ElevatorBookends>().startLeftBarrier == false)
+            {
+                theStartLeftBarrier.SetActive(false);
+            }
+            if (theStartPoint.gameObject.GetComponent<ElevatorBookends>().startRightBarrier == true)
+            {
+                theStartRightBarrier.SetActive(true);
+            }
+            else if (theStartPoint.gameObject.GetComponent<ElevatorBookends>().startRightBarrier == false)
+            {
+                theStartRightBarrier.SetActive(false);
+            }
+            if (theEndPoint.gameObject.GetComponent<ElevatorBookends>().endLeftBarrier == true)
+            {
+                theEndLeftBarrier.SetActive(true);
+            }
+            else if (theEndPoint.gameObject.GetComponent<ElevatorBookends>().endLeftBarrier == false)
+            {
+                theEndLeftBarrier.SetActive(false);
+            }
+            if (theEndPoint.gameObject.GetComponent<ElevatorBookends>().endRightBarrier == true)
+            {
+                theEndRightBarrier.SetActive(true);
+            }
+            else if (theEndPoint.gameObject.GetComponent<ElevatorBookends>().endRightBarrier == false)
+            {
+                theEndRightBarrier.SetActive(false);
+            }
+        }
+
+
+
+        if (waitCounter < 0f)
+        {
+            notMoving = false;
+            objectToMove.transform.position = Vector3.MoveTowards(objectToMove.transform.position, currentTarget, moveSpeed * Time.deltaTime);
```

Comment:
what is "maxDistanceDelta"?

Though I went and used time.fixeddeltatime instead